### PR TITLE
HDDS-9749. Infinite loop in ReconUtils.nextClosestPowerIndexOfTwo().

### PR DESCRIPTION
**Pull request by szetszwo**
## What changes were proposed in this pull request?

Fix the following method which does not work for 
- negative number (infinite loop), and
- positive power of two (incorrect value).
```java
//ReconUtils
  private static int nextClosestPowerIndexOfTwo(long dataSize) {
    int index = 0;
    while (dataSize != 0) {
      dataSize >>= 1;
      index += 1;
    }
    return index;
  }
```

## What is the link to the Apache JIRA

HDDS-9749

## How was this patch tested?

Added a new test.

### CODE REVIEW

**Comment by szetszwo** [comment]
@jojochuang , sure.  Added try-catch in `process(..)`.

File: hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/ReconUtils.java [hunk 1]
```diff
@@ -337,13 +342,11 @@ public static int getContainerSizeBinIndex(long containerSize) {
...
 
-  private static int nextClosestPowerIndexOfTwo(long dataSize) {
-    int index = 0;
-    while (dataSize != 0) {
-      dataSize >>= 1;
-      index += 1;
-    }
-    return index;
+  static int nextClosestPowerIndexOfTwo(long n) {
+    return n > 0 ? 64 - Long.numberOfLeadingZeros(n - 1)
```
Comment (jojochuang) [inline hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/ReconUtils.java:346]: genius

**Review by jojochuang — APPROVED** [review]

**Review by jojochuang — COMMENTED** [review]
Actually the test didn't pass TestContainerSizeCountTask.testProcess

**Review by jojochuang — APPROVED** [review]
LGTM

**Comment by szetszwo** [comment]
@jojochuang , thanks a lot for reviewing this!

**Pull request merged**